application/image/decorator/filter_decorators.hpp

```diff
@@ -0,0 +1,66 @@
+///
+/// @file filter_decorators.hpp
+/// @author fbtom
+/// @brief
+/// @date 2025-05-08
+///
+/// @copyright Copyright (c) 2025
+///
+
+#pragma once
+
+#include "application/image/image_process/image_processor.hpp"
+#include "opencv2/opencv.hpp"
+#include <memory>
+
+namespace image {
+namespace decorator {
+
+class FilterDecorator : public ImageProcessor {
+protected:
+  std::unique_ptr<ImageProcessor> image_;
```

Comment:
```suggestion
  std::unique_ptr<ImageProcessor> processor_;
```
or `image_processor_`?

Reply:
Variable name updated